lib/model/rofolder.go

```diff
@@ -16,113 +16,135 @@ import (
 
 type roFolder struct {
 	stateTracker
+	scan
```

Comment:
This is a fairly neat extraction, but the gains would be larger if it's reused by rwFolder as well.